### PR TITLE
Update from `cisagov/skeleton-python-library`

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request brings in upstream changes from [cisagov/skeleton-python-library] to bring this repository into our current development and testing practices.

## 💭 Motivation and context ##

Currently the GitHub Actions testing is not running in #5 and this is likely because of how out of date the configuration is in this project. Additionally if we are making changes to this project it should be brought up to date and we should ensure it will keep getting automatically updated with [cisagov/action-lineage] until such a time as we archive the project.

## 🧪 Testing ##

Automated tests pass. I coordinated with @king-alexander and he verified functionality with a branch using #5 rebased on this branch. He manually built a new Docker image, verified functionality, then built a new Docker image again this time with an empty `INSTALL_IPYTHON` to ensure updated logic worked with the other Dockerfile changes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

[cisagov/skeleton-python-library]: https://github.com/cisagov/skeleton-python-library
[cisagov/action-lineage]: https://github.com/cisagov/action-lineage

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
This all looks good to me, though maybe we should add an issue here to get the commented-out tests (the ones that require the Docker composition to start up) running again or else come up with other tests that can take their place.

File: tests/cert_test.py [hunk 1]
```diff
@@ -2,43 +2,48 @@
...
-        # get the first id from the summaries
-        id = summary.get()[0]["id"]
-        print(f"requesting certificate for id: {id}")
-        first_cert = cert_by_id.delay(id)
-        pem = first_cert.get(timeout=60)
-        print("done")
-
-        cert = x509.load_pem_x509_certificate(bytes(pem, "utf-8"), default_backend())
-        print(f"certificate serial number: {cert.serial_number}")
+# This test currently require the Docker Compose configuration to be running
```
Comment (jsf9k) [inline tests/cert_test.py]:
```suggestion
# This test currently requires the Docker Compose configuration to be running
```

**Review by jsf9k — APPROVED** [review]
Other than one tiny grammar thang in a comment, this all looks great to me.  Thanks for bringing this one into the 21st century!

**Comment by mcdonnnj** [comment]
> This all looks good to me, though maybe we should add an issue here to get the commented-out tests (the ones that require the Docker composition to start up) running again or else come up with other tests that can take their place.

@dav3r Please see #9 for tracking this super-duper high priority task 😏

**Comment by dav3r** [comment]
> > This all looks good to me, though maybe we should add an issue here to get the commented-out tests (the ones that require the Docker composition to start up) running again or else come up with other tests that can take their place.
> 
> @dav3r Please see #9 for tracking this super-duper high priority task 😏

Thanks @mcdonnnj for writing up that issue! 👍

**Comment by jsf9k** [comment]
@mcdonnnj, is this one ready to merge yet?  #5 needs these changes.

**Comment by mcdonnnj** [comment]
> @mcdonnnj, is this one ready to merge yet? #5 needs these changes.

@jsf9k I'm trying to coordinate with @king-alexander to test his code rebased on this to make sure nothing's been broken by me.

**Comment by mcdonnnj** [comment]
Re-requesting reviews because of the changes to `Dockerfile-admiral`.

**Review by jsf9k — APPROVED** [review]
_Approval intensifies_

File: tests/cert_test.py [hunk 2]
```diff
@@ -2,43 +2,48 @@
...
-        print(f"requesting certificate for id: {id}")
-        first_cert = cert_by_id.delay(id)
-        pem = first_cert.get(timeout=60)
-        print("done")
-
-        cert = x509.load_pem_x509_certificate(bytes(pem, "utf-8"), default_backend())
-        print(f"certificate serial number: {cert.serial_number}")
+# This test currently requires the Docker Compose configuration to be running
+# to function. This makes it difficult to perform standard Python testing.
+# class TestCerts:
```
Comment (dav3r) [inline tests/cert_test.py:30]:
```suggestion
# See https://github.com/cisagov/admiral/issues/9 for details.
# class TestCerts:
```

File: tests/scan_test.py [hunk 3]
```diff
@@ -22,22 +25,25 @@ def celery():
...
-        ns1 = up_scan.delay(host_ip)
-        PP.pprint(ns1.get())
-
-    def test_port_scan(self, celery, host_ip):
-        """Test port_scan task."""
-        ns2 = port_scan.delay(host_ip)
-        PP.pprint(ns2.get())
+# These tests currently require the Docker Compose configuration to be running
+# to function. This makes it difficult to perform standard Python testing.
+# class TestPortScans:
```
Comment (dav3r) [inline tests/scan_test.py:39]:
```suggestion
# See https://github.com/cisagov/admiral/issues/9 for details.
# class TestPortScans:
```

**Review by dav3r — APPROVED** [review]
👍 👍  Though I suggested inserting a couple of links to https://github.com/cisagov/admiral/issues/9 in comments.

**Pull request merged**